### PR TITLE
Refactor server.go file and add tests

### PR DESCRIPTION
**Pull request by andywaltlova**
Taken from #42 , just refactor with tests

### CODE REVIEW

File: src/server.go [hunk 1]
```diff
@@ -59,35 +93,12 @@ type jobServer struct {
...
-		commandOutput := processSignedScript(d.GetContent())
-
-		// Dial the Dispatcher and call "Finish"
-		conn, err := grpc.Dial(
-			yggdDispatchSocketAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
-		if err != nil {
-			log.Error(err)
-		}
-		defer conn.Close()
+func (s *jobServer) Send(_ context.Context, d *pb.Data) (*pb.Receipt, error) {
```
Comment (r0x0d) [inline src/server.go:96]: Since we are not using context anymore, do we still need it here?

**Review by r0x0d — COMMENTED** [review]
Looks good, just one question about the context usage in the server.go file

File: src/server.go (hunk 1 above)
Comment (andywaltlova) [reply src/server.go:96]: no we don't, even before we were just reassigning it with new context but golang will complain about wrong method if I remove it :) 

image

Comment (r0x0d) [reply src/server.go:96]: Aha, I see. You're right! 

Thanks for the clarification.

**Review by r0x0d — APPROVED** [review]

**Pull request merged**